### ISSUE TITLE
Remove AutomaticBean

### ISSUE THREAD

**Issue by stoyanK7** [checkstyle/checkstyle#12873]
From https://github.com/checkstyle/checkstyle/pull/12213#discussion_r1142109362
> lets create issue to remove it in 2025 or Checkstyle 11

---
After #12213 gets merged, `AutomaticBean` will be deprecated and replaced with `AbstractAutomaticBean`. After some time `AutomaticBean` has to be removed.

---
We keep `AutomaticBean` because maven checkstyle plugin relies on it as shown in https://github.com/checkstyle/checkstyle/pull/12213#issuecomment-1253907534.

**Comment by Boukhalfa11** [comment]
Hi, can I  work in this issue please ?

**Comment by romani** [comment]
Issue is not approved for now on purpose

**Comment by michael-o** [comment]
Folks, I trying currently to reconciple Maven Checkstyle Plugin which depends on 9.3 (and cannot be upgraded ATM) for those who want to swap to 10.x at runtime. I get that dreaded runtime issue.
Is there anything me or you can do to solve the issue for the time being?
https://github.com/apache/maven-checkstyle-plugin/blob/c6dbdfea723968428ba3ed23bc88e5aad3b396f2/src/main/java/org/apache/maven/plugins/checkstyle/AbstractCheckstyleReport.java#L37 and https://github.com/apache/maven-checkstyle-plugin/blob/c6dbdfea723968428ba3ed23bc88e5aad3b396f2/src/main/java/org/apache/maven/plugins/checkstyle/AbstractCheckstyleReport.java#L619-L625

**Comment by romani** [comment]
> I get that dreaded runtime issue.

Please share link to runtime execution failure.
Compilation problem, our mistake, might be quicker to resolve by extra conversation lines https://github.com/apache/maven-checkstyle-plugin/pull/141#discussion_r1630515144

**Comment by michael-o** [comment]
> > I get that dreaded runtime issue.
> 
> Please share link to runtime execution failure. Compilation problem, our mistake, might be quicker to resolve by extra conversation lines [apache/maven-checkstyle-plugin#141 (comment)](https://github.com/apache/maven-checkstyle-plugin/pull/141#discussion_r1630515144)

https://github.com/apache/maven-checkstyle-plugin/pull/136#issuecomment-2152517655

**Comment by romani** [comment]
Yes, better to make plugin version that supports checkstyle 10.0 until 10.9.2, and one more release that support 10.9.3 and up
